app-libs/exchange-oracle/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(test, feature(assert_matches))]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+#[macro_use]
+extern crate sgx_tstd as std;
+
+// re-export module to properly feature gate sgx and regular std environment
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub mod sgx_reexport_prelude {
+	pub use thiserror_sgx as thiserror;
+	pub use url_sgx as url;
+}
+
+use crate::error::Error;
+
+pub mod coingecko;
+pub mod error;
+
+pub trait GetExchangeRate {
+	//Get exchange rate of vs_currency to coin ( = coin/vs_currency. Ex: DOT/usd )
```

Comment:
Sorry this comment is completely wrong. It should be
 `Get the exchange rate coin to currency ( rate = coin/currency)`
 The vs_currency is a "relic" from the API "coingecko". I will rename it to currency.
Your examples strings are specific to coingecko API. I don't think they should be added here. What about:
 ```
    /// Get the exchange rate coin to currency ( rate = coin/currency)
    /// coin: a cryptocurrency
    /// currency: a fiat currency
    ///fn get_exchange_rate(&mut self, coin: &str, currency: &str) -> Result<f32, Error>;
 ```

Reply:
Much better! There's an `of` missing in the first sentence: -> ` Get the exchange rate of coin to currency (rate = coin/currency).`
And instead of writing it in the comment, maybe replace the variable names all together to `cryptocurrency` & `fiat_currency`?